src/game_api/screen.hpp

```diff
@@ -226,79 +312,90 @@ class ScreenOptions : public Screen // ID: 5
     TextureRenderingInfo item_option_arrow_right;
     TextureRenderingInfo tooltip_background;
     TextureRenderingInfo progressbar_background; // brightness 'progressbar' background texture
-    TextureRenderingInfo unknown40;
-    TextureRenderingInfo progressbar_foreground;
-    TextureRenderingInfo progressbar_position_indicator;
-    TextureRenderingInfo sectionheader_background; // behind 'GRAPHICS' and 'AUDIO'
+    TextureRenderingInfo volume_progressbar_foreground;
+    TextureRenderingInfo progressbar_foreground; // the border
+    TextureRenderingInfo volume_progressbar_position_indicator;
+    TextureRenderingInfo sectionheader_background; // behind 'GRAPHICS' and 'AUDIO' the black bars
 
-    TextureRenderingInfo unknown44;
+    TextureRenderingInfo unknown44; // could be something else, or just not used in PC version or something
     TextureRenderingInfo unknown45;
-    float topleft_woodpanel_esc_slidein_timer;
-    float text_fadein_timer;
-    float vertical_scroll_effect_timer;
+    TextureRenderingInfo unknown46;
+    TextureRenderingInfo unknown47;
+    TextureRenderingInfo unknown48;
+    float unknown48a; // was supposto be topleft_woodpanel_esc_slidein?
+    float text_fadein;
+    float vertical_scroll_effect;
     uint8_t unknown49;
-    uint8_t unknown50;
-    uint8_t unknown51;
+    bool item_visiable;
+    bool item_highlight;
+    // uint8_t padding_probably8[5];
+    custom_vector<size_t> unknown50; // holds one 8 byte value, related to choosen opion menu
+    uint8_t unknown51;               // probably bool
+    // padding_probably10[7];
+    custom_vector<STRINGID> tooltip_text;
+    size_t unknown53;
+    uint32_t unknown54; // some timer
+    uint32_t unknown55; // probably padding
+    size_t* credits_related;
+    size_t unknown57;
 };
 
-class ScreenPlayerProfile : public Screen // ID: 6
-{
-  public: // not reverse engineered
-};
-
-class ScreenLeaderboards : public Screen // ID: 7
-{
-  public: // not reverse engineered
-};
-
-class ScreenSeedInput : public Screen // ID: 8
+class ScreenCodeInput : public Screen // ID: 8
 {
   public:
-    uint16_t unknown4;
-    uint16_t unknown5;
-    float bottom_woodpanel_slideup_timer;
-    float unknown_timer;
-    float unknown6;
-    float bottom_woodpanel_y;
-    TextureRenderingInfo bottom_woodpanel;
-    TextureRenderingInfo unknown8;
-    TextureRenderingInfo unknown9;
-    TextureRenderingInfo unknown10;
-    TextureRenderingInfo unknown11;
-    STRINGID unknown12;
-    STRINGID unknown13;
-    STRINGID buttons_text_id;
-    STRINGID unknown15;
-    uint8_t unknown16;
-    uint8_t unknown17;
-    uint8_t unknown18;
-    uint8_t unknown19;
-    uint8_t unknown20;
-    uint8_t unknown21;
-    uint8_t unknown22;
-    uint8_t unknown23;
-    uint32_t unknown24;
-    uint16_t unknown25;
+    // This beginning is the same for ScreenOnlineLobby and similar code can be found in ScreenOptions
```

Comment:
Was thinking of making this separate struct, but can't think of a name for it
It's two wooden panels at the top and bottom with the text and all

Reply:
I guess it would be a Headers or Border or Panels